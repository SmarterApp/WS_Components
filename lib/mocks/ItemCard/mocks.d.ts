import { ItemCardModel } from "@src/index";
export declare const completeItemCard: ItemCardModel;
export declare const itemCardList: ItemCardModel[];
export declare const sortableItemCards: ItemCardModel[];
