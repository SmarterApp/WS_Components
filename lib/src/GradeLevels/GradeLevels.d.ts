export declare enum GradeLevels {
    NA = 0,
    Grade3 = 1,
    Grade4 = 2,
    Grade5 = 4,
    Grade6 = 8,
    Grade7 = 16,
    Grade8 = 32,
    Grade9 = 64,
    Grade10 = 128,
    Grade11 = 256,
    Grade12 = 512,
    Elementary = 7,
    Middle = 56,
    High = 960,
    All = 1023,
}
export declare class GradeLevel {
    static gradeCaseToString(grade: GradeLevels): string | undefined;
    static gradeCaseToShortString(grade: GradeLevels): string | undefined;
    static gradeLevelToString(grades: GradeLevels): string | undefined;
    static gradeLevelContains(haystack: GradeLevels, needle: GradeLevels): boolean;
    static stringToGradeLevel(gradeString: string): GradeLevels;
    static gradeLevelAdd(grade: GradeLevels, gradeString: string): number;
}
