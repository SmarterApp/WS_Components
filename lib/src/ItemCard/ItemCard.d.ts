import * as React from "react";
import { ItemCardModel } from "./ItemCardModels";
export interface ItemCardState {
    redirect: boolean;
}
export declare class ItemCard extends React.Component<ItemCardModel, ItemCardState> {
    constructor(props: ItemCardModel);
    shouldComponentUpdate(nextProps: ItemCardModel, nextState: ItemCardState): boolean;
    handleKeyPress: (e: React.KeyboardEvent<HTMLElement>) => void;
    handleOnClick: () => void;
    render(): JSX.Element;
}
