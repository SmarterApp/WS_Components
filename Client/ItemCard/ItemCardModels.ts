import * as GradeLevels from '../GradeLevels/GradeLevels';
import { Link } from 'react-router-dom';

export function itemPageLink(bankKey: number, itemKey: number) {
    window.location.href = "/Item?bankKey=" + bankKey + "&itemKey=" + itemKey;
}

export interface ItemCardViewModel {
    bankKey: number;
    itemKey: number;
    title: string;
    grade: GradeLevels.GradeLevels;
    gradeLabel: string;
    subjectCode: string;
    subjectLabel: string;
    claimCode: string;
    claimLabel: string;
    targetHash: string;
    targetShortName: string;
    interactionTypeCode: string;
    interactionTypeLabel: string;
}
