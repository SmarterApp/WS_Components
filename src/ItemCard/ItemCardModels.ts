import * as GradeLevels from '../GradeLevels/GradeLevels';
import { Link } from 'react-router-dom';

export function itemPageLink(bankKey: number, itemKey: number) {
    window.location.href = "/Item?bankKey=" + bankKey + "&itemKey=" + itemKey;
}

export interface ItemCardModel {
    bankKey: number;
    itemKey: number;
    title: string;
    grade: GradeLevels.GradeLevels;
    gradeLabel: string;
    subjectCode: string;
    subjectLabel: string;
    claimCode: string;
    claimLabel: string;
    targetHash: number;
    targetShortName: string;
    interactionTypeCode: string;
    interactionTypeLabel: string;
    isPerformanceItem: boolean;
}
