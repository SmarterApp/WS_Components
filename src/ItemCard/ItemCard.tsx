import * as React from "react";
import * as GradeLevels from "../GradeLevels/GradeLevels";
import * as ItemCardModels from "./ItemCardModels";
import { Redirect } from "react-router";

export interface ItemCardState {
  redirect: boolean;
}
export interface ItemCardProps extends ItemCardModels.ItemCardModel {}

export class ItemCard extends React.Component<ItemCardProps, ItemCardState> {
  constructor(props: ItemCardProps) {
    super(props);
    this.state = { redirect: false };
  }

  shouldComponentUpdate(nextProps: ItemCardProps, nextState: ItemCardState) {
    return nextState.redirect;
  }

  handleKeyPress = (e: React.KeyboardEvent<HTMLElement>) => {
    if (e.keyCode === 13 || e.keyCode === 23) {
      this.setState({ redirect: true });
    }
  };

  handleOnClick = () => {
    this.setState({ redirect: true });
  };

  render() {
    const { bankKey, itemKey } = this.props;
    let content = <Redirect push to={`/Item/${bankKey}-${itemKey}`} />;
    if (!this.state.redirect) {
      content = (
        <div
          role="link"
          className={`card card-block ${this.props.subjectCode.toLowerCase()}`}
          onClick={this.handleOnClick}
          onKeyUp={this.handleKeyPress}
          tabIndex={0}
        >
          <div className="card-contents">
            <h4 className="card-title">{this.props.title}</h4>
            <p className="card-text subject">
              <span className="card-text-label">Subject:</span>
              <span className="card-text-value">
                {" "}
                {this.props.subjectLabel}
              </span>
            </p>
            <p className="card-text grade">
              <span className="card-text-label">Grade:</span>
              <span className="card-text-value"> {this.props.gradeLabel}</span>
            </p>
            <p className="card-text claim">
              <span className="card-text-label">Claim:</span>
              <span className="card-text-value"> {this.props.claimLabel}</span>
            </p>
            <p className="card-text target">
              <span className="card-text-label">Target:</span>
              <span className="card-text-value">
                {this.props.targetShortName}
              </span>
            </p>
            <p className="card-text interaction-type">
              <span className="card-text-label">Item Type:</span>
              <span className="card-text-value">
                {` ${this.props.interactionTypeLabel}`}
              </span>
            </p>
            <p className="card-text item-id">
              <span className="card-text-label">Item Id:</span>
              <span className="card-text-value"> {this.props.itemKey}</span>
            </p>
          </div>
        </div>
      );
    }

    return content;
  }
}
