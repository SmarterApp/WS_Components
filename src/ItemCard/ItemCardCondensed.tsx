import "src/Assets/Styles/item-card.less";
import * as React from "react";
import * as ItemCardModels from "./ItemCardModels";
import { Redirect } from "react-router";

export interface ItemCardCondensedState {
  redirect: boolean;
}
export interface ItemCardCondensedProps extends ItemCardModels.ItemCardModel {}

export class ItemCardCondensed extends React.Component<
  ItemCardCondensedProps,
  ItemCardCondensedState
> {
  constructor(props: ItemCardCondensedProps) {
    super(props);
    this.state = { redirect: false };
  }

  shouldComponentUpdate(
    nextProps: Readonly<ItemCardCondensedProps>,
    nextState: Readonly<ItemCardCondensedState>,
    nextContext: any
  ): boolean {
    return this.state.redirect;
  }

  handleKeyPress(e: React.KeyboardEvent<HTMLElement>) {
    if (e.keyCode === 13 || e.keyCode === 23) {
      this.setState({ redirect: true });
    }
  }
  handleOnClick() {
    this.setState({ redirect: true });
  }

  render() {
    const { bankKey, itemKey } = this.props;
    if (this.state.redirect) {
      return <Redirect push to={`/Item/${bankKey}-${itemKey}`} />;
    }
    return (
      <div
        className={`card card-block ${this.props.subjectCode.toLowerCase()} condensed`}
        onClick={() => this.handleOnClick()}
        onKeyUp={e => this.handleKeyPress(e)}
        tabIndex={0}
      >
        <div className="card-contents">
          <h4 className="card-title">{this.props.subjectLabel}</h4>
          <p className="card-text claim">
            <span className="card-text-label">Claim:</span>
            <span className="card-text-value"> {this.props.claimLabel}</span>
          </p>
          <p className="card-text target">
            <span className="card-text-label">Target:</span>
            <span className="card-text-value">
              {" "}
              {this.props.targetShortName}
            </span>
          </p>
          <p className="card-text interaction-type">
            <span className="card-text-label">Item Type:</span>
            <span className="card-text-value">
              {" "}
              {this.props.interactionTypeLabel}
            </span>
          </p>
          <p className="card-text item-id">
            <span className="card-text-label">Item Id:</span>
            <span className="card-text-value"> {this.props.itemKey}</span>
          </p>
        </div>
      </div>
    );
  }
}
