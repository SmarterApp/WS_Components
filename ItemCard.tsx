import * as React from 'react';
import * as GradeLevels from './GradeLevels';
import * as API from './ApiModels';
import * as AboutItem from './AboutItem';
import * as ItemCardViewModel from './ItemCardViewModel'
import * as ItemCardFields from './ItemCardFields'
import * as ItemCardCondensed from './ItemCardCondensed'

export interface Props{
    aboutItem: ItemCardViewModel.ItemCardViewModel;
}

export class ItemCard extends React.Component<Props, {}> {
    handleKeyPress(bankKey: number, itemKey: number, e: React.KeyboardEvent<HTMLElement>) {
        if (e.keyCode === 13 || e.keyCode === 23) {
            ItemCardFields.itemPageLink(bankKey, itemKey);
        }
    }

    shouldComponentUpdate(nextProps: Readonly<Props>, nextState: Readonly<{}>, nextContext: any): boolean {
        return false;
    }

    renderCard(bankKey: number, itemKey: number) {
        return (
            <div className={`card card-block ${this.props.aboutItem.subjectCode.toLowerCase()}`}
                onClick={e => ItemCardFields.itemPageLink(bankKey, itemKey)}
                onKeyUp={e => this.handleKeyPress(bankKey, itemKey, e)}
                tabIndex={0}>
                <div className="card-contents">
                    <ItemCardFields.ItemCardFields aboutItem={this.props.aboutItem}/>
                </div>
            </div>
        );
    }

    render() {
        return (
            <div>
                {this.renderCard(this.props.aboutItem.bankKey, this.props.aboutItem.itemKey)};
            </div>
        );
    }
}