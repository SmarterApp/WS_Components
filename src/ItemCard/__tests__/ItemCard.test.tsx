import * as React from "react";
import { shallow } from "enzyme";

import { ItemCard, ItemCardProps } from "../ItemCard";
import { GradeLevels } from "../../GradeLevels/GradeLevels";
import { Redirect } from "react-router";

const itemVM: ItemCardProps = {
  bankKey: 1,
  itemKey: 1,
  gradeLabel: "grade 3",
  subjectLabel: "math",
  claimCode: "234536",
  targetId: "23463467",
  domain: "MATH",
  depthOfKnowledge: "",
  commonCoreStandardId: "",
  title: "",
  grade: GradeLevels.Grade6,
  subjectCode: "MATH",
  claimLabel: "Math Claim",
  targetShortName: "1-3",
  interactionTypeCode: "",
  interactionTypeLabel: "",
  isPerformanceItem: false,
  brailleOnlyItem: false,
  targetHash: 1212
};

describe("ItemCard", () => {
  const wrapper = shallow(<ItemCard {...itemVM} />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: test redirects
});
