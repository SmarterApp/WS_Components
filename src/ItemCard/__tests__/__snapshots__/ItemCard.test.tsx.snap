// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<div
  className="card card-block math"
  onClick={[Function]}
  onKeyUp={[Function]}
  tabIndex={0}
>
  <div
    className="card-contents"
  >
    <h4
      className="card-title"
      onClick={[Function]}
      onKeyUp={[Function]}
    />
    <p
      className="card-text subject"
    >
      <span
        className="card-text-label"
      >
        Subject:
      </span>
      <span
        className="card-text-value"
      >
         
        math
      </span>
    </p>
    <p
      className="card-text grade"
    >
      <span
        className="card-text-label"
      >
        Grade:
      </span>
      <span
        className="card-text-value"
      >
         
        grade 3
      </span>
    </p>
    <p
      className="card-text claim"
    >
      <span
        className="card-text-label"
      >
        Claim:
      </span>
      <span
        className="card-text-value"
      >
         
        Math Claim
      </span>
    </p>
    <p
      className="card-text target"
    >
      <span
        className="card-text-label"
      >
        Target:
      </span>
      <span
        className="card-text-value"
      >
        1-3
      </span>
    </p>
    <p
      className="card-text interaction-type"
    >
      <span
        className="card-text-label"
      >
        Item Type:
      </span>
      <span
        className="card-text-value"
      >
         
      </span>
    </p>
    <p
      className="card-text item-id"
    >
      <span
        className="card-text-label"
      >
        Item Id:
      </span>
      <span
        className="card-text-value"
      >
         
        1
      </span>
    </p>
  </div>
</div>
`;
